datafusion/core/tests/sql/create_drop.rs

```diff
@@ -15,427 +15,11 @@
-    ctx.sql(sql).await.unwrap();
-
-    let sql_all = "SELECT * FROM my_table";
-    let results_all = execute_to_batches(&ctx, sql_all).await;
-
-    let expected = vec![
-        "+----+----+-------+",
-        "| c1 | c2 | c3    |",
-        "+----+----+-------+",
-        "| 1  | 2  | hello |",
-        "+----+----+-------+",
-    ];
-
-    assert_batches_eq!(expected, &results_all);
-
-    Ok(())
-}
-
-#[tokio::test]
-async fn create_or_replace_table_as() -> Result<()> {
-    // the information schema used to introduce cyclic Arcs
-    let ctx =
-        SessionContext::with_config(SessionConfig::new().with_information_schema(true));
-
-    // Create table
-    let result = ctx
-        .sql("CREATE TABLE y AS VALUES (1,2),(3,4)")
-        .await
-        .unwrap()
-        .collect()
-        .await
-        .unwrap();
-    assert!(result.is_empty());
-
-    // Replace table
-    ctx.sql("CREATE OR REPLACE TABLE y AS VALUES (5,6)")
-        .await
-        .unwrap()
-        .collect()
-        .await
-        .unwrap();
-
-    let sql_all = "SELECT * FROM y";
-    let results_all = execute_to_batches(&ctx, sql_all).await;
-
-    let expected = vec![
-        "+---------+---------+",
-        "| column1 | column2 |",
-        "+---------+---------+",
-        "| 5       | 6       |",
-        "+---------+---------+",
-    ];
-
-    assert_batches_eq!(expected, &results_all);
-
-    // 'IF NOT EXISTS' cannot coexist with 'REPLACE'
-    let result = ctx
-        .sql("CREATE OR REPLACE TABLE if not exists y AS VALUES (7,8)")
-        .await;
-    assert!(
-        result.is_err(),
-        "'IF NOT EXISTS' cannot coexist with 'REPLACE'"
-    );
-
-    Ok(())
-}
-
-#[tokio::test]
-async fn drop_table() -> Result<()> {
-    let ctx = SessionContext::new();
-    register_aggregate_simple_csv(&ctx).await?;
-
-    let sql = "CREATE TABLE my_table AS SELECT * FROM aggregate_simple";
-    ctx.sql(sql).await.unwrap();
-
-    let sql = "DROP TABLE my_table";
-    ctx.sql(sql).await.unwrap();
-
-    let result = ctx.table("my_table");
-    assert!(result.is_err(), "drop table should deregister table.");
-
-    let sql = "DROP TABLE IF EXISTS my_table";
-    ctx.sql(sql).await.unwrap();
-
-    Ok(())
-}
-
-#[tokio::test]
-async fn drop_view() -> Result<()> {
-    let ctx =
-        SessionContext::with_config(SessionConfig::new().with_information_schema(true));
-    plan_and_collect(&ctx, "CREATE VIEW v AS SELECT 1").await?;
-    let rb = plan_and_collect(
-        &ctx,
-        "select * from information_schema.tables where table_name = 'v' and table_type = 'VIEW'",
-    )
-    .await?;
-    assert_eq!(rb[0].num_rows(), 1);
-
-    plan_and_collect(&ctx, "DROP VIEW v").await?;
-    let rb = plan_and_collect(
-        &ctx,
-        "select * from information_schema.tables where table_name = 'v' and table_type = 'VIEW'",
-    )
-    .await?;
-    assert!(rb.is_empty());
-    Ok(())
-}
-
-#[tokio::test]
-#[should_panic(expected = "doesn't exist")]
```

Comment:
One difference is that .slt files don't seem to offer a way to expect a certain error message (seemingly by design). I am not sure if you have any thoughts / experience on this @xudong963

Reply:
I think it can be supported in sqllogictest-rs. 

Currently, databend has written its own python version of the logictest framework, which is able to add error messages, such as `statement error 1006`: 1006 is errorcode.

So I think it's a reasonable feature ~  ping @xxchan.

Reply:
We do support error messages:

`statement error <regexp>` and `query error <regexp>`

example can be found here https://github.com/risinglightdb/sqllogictest-rs/blob/main/examples/basic/basic.slt#L20

Reply:
Sweet -- hank you @xxchan  I will update these tests to use that syntax

Reply:
I added the expected messages in 6be3662b6 -- very cool! It was good enough that I see a few bugs

Reply:
Here is an improvement to fix the messages: https://github.com/apache/arrow-datafusion/pull/4572